### PR TITLE
Do FIB lookup on to-host path.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Looks like a bug that we were doing it on the opposite path.
## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: bpf-gpl/tc.c [hunk 1]
```diff
@@ -586,9 +586,9 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 	}
 
 	/* We are possibly past (D)NAT, but that is ok, we need to let the IP
-	 * stack do the RPF check on the source, dest is not importatnt.
+	 * stack do the RPF check on the source, dest is not important.
 	 */
-	if (CALI_F_TO_HOST && ct_result_rpf_failed(state.ct_result.rc)) {
```
Comment (tomastigera) [inline bpf-gpl/tc.c:591]: The tiny benefit of the `CALI_F_TO_HOST &&` is that it is compiled out for other programs. Probably makes little difference in runtime and the code is simpler without it.

File: bpf-gpl/tc.c [hunk 2]
```diff
@@ -801,12 +801,14 @@ static CALI_BPF_INLINE struct fwd calico_tc_skb_accepted(struct __sk_buff *skb,
...
 	if (CALI_F_FROM_WEP && (state->flags & CALI_ST_NAT_OUTGOING)) {
+		// We are going to SNAT this traffic, using iptables SNAT so set the mark
+		// to trigger that and leave the fib lookup disabled.
 		seen_mark = CALI_SKB_MARK_NAT_OUT;
 	} else {
-		/* XXX we do it here again because doing it in one place only
-		 * XXX in calico_tc() irritates the verifier :'(
-		 */
-		if (!CALI_F_TO_HOST || !ct_result_rpf_failed(state->ct_result.rc)) {
+		// Non-SNAT case, see if we can do the FIB lookup.  Note: tried to pass in
```
Comment (tomastigera) [inline bpf-gpl/tc.c:808]:
```suggestion
		// Non-SNAT case, allow FIB lookup only if RPF check passed.  Note: tried to pass in
```

File: bpf-gpl/tc.c (hunk 1 above)
Comment (fasaxc) [reply bpf-gpl/tc.c:591]: I'm hoping the compiler is smart enough to spot that fib is already set to false at the top so this conditional is a no-op.

**Comment by fasaxc** [comment]
Think @tomastigera was happy with this and CI fail was a semaphore funny (on an unrelated code path).

**Pull request merged**